src/components/common/modal/index.tsx

```diff
@@ -1,7 +1,41 @@
 import React from "react";
+import { useAppDispatch, useAppSelector } from "@/state/app/hooks";
+import { handleModal } from "@/state/features";
+import { Icon, ContactForm } from "@/components/common";
+
+import styles from "./Modal.module.css";
 
 const Modal = () => {
-  return <div>Modal</div>;
+  const dispatch = useAppDispatch();
+  const { modalOpen } = useAppSelector((state) => state.modal);
+
+  const closeModal = () => {
+    dispatch(handleModal(false));
+  };
+
+  return (
+    <>
+      <div
+        className={`${styles.overlay} center flex-column ${
+          modalOpen ? styles.opened : styles.closed
+        }`}
+        onClick={closeModal}></div>
```

Comment:
Do you have prettier (code formatter) configured? It's weird how is closing this `div`

Reply:
I do! It's weird tho, I believe it's because it's empty.
It lets me adjust it only if I keep an empty string inside, but I think that looks even weirder

```suggestion
      <div
        className={`${styles.overlay} center flex-column ${
          modalOpen ? styles.opened : styles.closed
        }`}
        onClick={closeModal}>
        {""}
        </div>
```

Reply:
You can use one self closed tag, like `<div />`